### PR TITLE
Initialize `isNativePreviewSidebarOpen` as false if it's `undefined`

### PR DESCRIPTION
**Pull request by nemanjaglumac**
This PR makes sure that the redux `qb.uiControls.isNativePreviewSidebarOpen` setting is initialized as `false` even when it is initially `undefined`, i.e. before the user explicitly sets its value.

### CODE REVIEW

File: frontend/src/metabase/query_builder/actions/core/initializeQB.unit.spec.ts [hunk 1]
```diff
@@ -832,5 +832,10 @@ describe("QB Actions > initializeQB", () => {
       });
       expect(result.uiControls.isShowingNewbModal).toBeFalsy();
     });
+
+    it("initializes `isNativePreviewSidebarOpen` value as false when it is undefined", async () => {
+      const { result } = await setupOrdersTable();
+      expect(result.uiControls.isNativePreviewSidebarOpen).toBe(false);
+    });
```
Comment (nemanjaglumac) [inline frontend/src/metabase/query_builder/actions/core/initializeQB.unit.spec.ts:839]: This test is just recycling the existing setup. There is no other reason for it to live under this specific `describe` block. This setting does not apply to any table, question or a model specifically. It is "global".

LMK if there's a better place for this test.

**Review by ranquild — COMMENTED** [review]
@nemanjaglumac could you use `DEFAULT_UI_CONTROLS` and set `isNativePreviewSidebarOpen: false` there? https://github.com/metabase/metabase/blob/2aa3cec610588238132afbc789da0845c7c2ec5d/frontend/src/metabase/query_builder/reducers.js#L73

**Review by uladzimirdev — APPROVED** [review]

**Comment by nemanjaglumac** [comment]
> @nemanjaglumac could you use `DEFAULT_UI_CONTROLS` and set `isNativePreviewSidebarOpen: false` there?
> 
> https://github.com/metabase/metabase/blob/2aa3cec610588238132afbc789da0845c7c2ec5d/frontend/src/metabase/query_builder/reducers.js#L73

@ranquild Sure. Here it is in https://github.com/metabase/metabase/pull/40912/commits/858d3586088ef2c4dae44583871435742e28d712.

I don't understand why is the unit test failing, and why the QB initializes this setting as `undefined` if I've provided a default value? Type check also fails if I set the type to only `boolean` instead of `boolean | undefined`

**Comment by ranquild** [comment]
@nemanjaglumac you also need to set the value in our mock state helpers for the QB. Then you would be able to make it just Boolean

**Comment by nemanjaglumac** [comment]
> @nemanjaglumac you also need to set the value in our mock state helpers for the QB. Then you would be able to make it just Boolean

That fixed the type issue. There are still two problems, though.

One is that the test fails because it still finds this setting `undefined` instead of `false`. 

```
  ● QB Actions > initializeQB › blank question › initializes `isNativePreviewSidebarOpen` value as false when it is undefined

    expect(received).toBe(expected) // Object.is equality

    Expected: false
    Received: undefined
```

And the other one is that the `metabase/qb/INITIALIZE_QB` now overrides user setting every time. It seems I'll have to bring back some sort of conditional logic, which is what I started this PR from.

**Comment by nemanjaglumac** [comment]
This might not even be needed after #41064

**Comment by nemanjaglumac** [comment]
Closing in favor of #41064 where this scenario is handled.

**Pull request closed without merging**